lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -628,4 +552,109 @@ function setupTransport(transport) {
   }
 }
 
+/**
+ * Create a Connect, Sync or Disconnect RSP message.
+ * @private
+ * @param {TwilioConnectionTransport} transport
+ * @param {string} accessToken
+ * @returns {?object}
+ */
+function createConnectOrSyncOrDisconnectMessage(transport, accessToken) {
+  const {
+    _adaptiveSimulcast,
+    _automaticSubscription,
+    _bandwidthProfile,
+    _dominantSpeaker,
+    _iceServersStatus,
+    _localParticipant,
+    _name,
+    _networkQuality,
+    _options,
+    _peerConnectionManager,
+    _renderHints,
+    _session,
+    _trackPriority,
+    _trackSwitchOff,
+    _userAgent,
+    state
+  } = transport;
+
+  if (state === 'connected') {
+    return null;
+  }
+
+  if (state === 'disconnected') {
+    return {
+      session: _session,
+      type: 'disconnect',
+      version: RSP_VERSION
+    };
+  }
+
+  const type = {
+    connecting: 'connect',
+    syncing: 'sync'
+  }[state];
+
+  const message = {
+    name: _name,
+    participant: _localParticipant.getState(),
+    peer_connections: _peerConnectionManager.getStates(),
+    type,
+    version: RSP_VERSION
+  };
+
+  if (message.type === 'connect') {
+    message.ice_servers = _iceServersStatus;
+
+    message.publisher = {
+      name: SDK_NAME,
+      sdk_version: SDK_VERSION,
+      user_agent: _userAgent
+    };
+
+    if (_bandwidthProfile) {
+      message.bandwidth_profile = createBandwidthProfilePayload(_bandwidthProfile);
+    }
+
+    message.media_signaling = createMediaSignalingPayload(
+      _dominantSpeaker,
+      _networkQuality,
+      _trackPriority,
+      _trackSwitchOff,
+      _adaptiveSimulcast,
+      _renderHints);
+
+    message.subscribe = createSubscribePayload(_automaticSubscription);
+
+    const { sdpFormat } = _options;
+    if (sdpFormat) {
+      message.format = sdpFormat;
+    }
+    message.token = accessToken;
```

Comment:
@manjeshbhargav, Can you give us some context on the purpose of this change. If it's about hiding access token from the user, wouldn't they be able to see it from the network tab anyways?

Reply:
@makarandp0 Since we are not charging for the Media Extension Participant minutes, there is a danger that the Custom Media Extension code, which is managed by the developer, can access this token and use it to join Rooms for free.

Reply:
@makarandp0 The network tab thing won't be a problem here since the media extension is running in a CEF environment on Twilio's cloud infrastructure.

Reply:
Do we have safeguards on the backend in case the token is used outside of the media extension participant?

Reply:
how does the Media Extension Participant get this access token in the 1st place - That is how do they call connect?

Reply:
@charliesantos @makarandp0 I believe the token is internally acquired by the MediaProcessor service and passed to the CEF context, where the token is used by an internal JS function to join a Room and then the Room object is passed to the application code. That is why we don't want any reference to the token in the Room object.